src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,169 @@
+
 package core.basesyntax;
 
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
+    }
+
+    private static class Node<T> {
+        private T data;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(T data) {
+            this.data = data;
+        }
+    }
+
+    private boolean checkIndex(int index) {
+        if (index >= 0 && index < size) {
+            return true;
+        } else {
+            throw new IndexOutOfBoundsException("Wrong index" + index);
+        }
+    }
+
+    private Node<T> findIndex(int index) {
+        Node<T> current = first;
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
+        return current;
+    }
+
+    private void removeLogic() {
+        Node<T> current = first;
+        if (current == first) {
```

Comment:
```suggestion
        Node<T> current = first;
        if (current == first) {
```

тут ты сначала приравниваешь current к first, а потом сравниваешь current и first? Проверь работу этого метода. Сюда на вход должна приходить нода, которую ты хочешь отлинковть